src/main/java/org/springframework/data/redis/connection/jedis/JedisClusterListCommands.java

```diff
@@ -361,27 +364,48 @@ public Long lRem(byte[] key, long count, byte[] value) {
 
 	/*
 	 * (non-Javadoc)
-	 * @see org.springframework.data.redis.connection.RedisListCommands#bLPop(int, byte[][])
+	 * @see org.springframework.data.redis.connection.RedisListCommands#rPop(byte[], long)
 	 */
 	@Override
 	public List<byte[]> bLPop(int timeout, byte[]... keys) {
+		return bLPop(timeout, TimeUnit.SECONDS, keys);
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.redis.connection.RedisListCommands#bLPop(int, java.util.concurrent.TimeUnit, byte[][])
+	 */
+	@Override
+	public List<byte[]> bLPop(int timeout, TimeUnit unit, byte[]... keys) {
 
 		Assert.notNull(keys, "Key must not be null!");
 		Assert.noNullElements(keys, "Keys must not contain null elements!");
 
 		if (ClusterSlotHashUtil.isSameSlotForAllKeys(keys)) {
 			try {
-				return connection.getCluster().blpop(timeout, keys);
+				if (TimeUnit.MILLISECONDS == unit) {
```

Comment:
Note that one could use `MICROSECONDS` and `NANOSECONDS` as units. Ideally, we add a utility method to `TimeoutUtils` along the lines of `supportsDoubleSeconds` or `isSubSecondUnit` instead of repeating the same code to check the timeout across the code base.